contracts/core/LensHub.sol

```diff
@@ -534,9 +532,13 @@ contract LensHub is ILensHub, LensNFTBase, VersionedInitializable, LensMultiStat
         override
         whenNotPaused
     {
-        bytes32[] memory dataHashes = new bytes32[](vars.datas.length);
-        for (uint256 i = 0; i < vars.datas.length; ++i) {
+        uint256 dataLength = vars.datas.length;
```

Comment:
is not more expensive than taking the length from calldata ?

Reply:
Nope, its' cheaper to cache it and prevent needing to re-read from calldata.